gitmanager.py

```diff
@@ -306,20 +306,56 @@ def remove_from_blacklist(cls, item, username, blacklist_type="", code_privilege
     @classmethod
     def merge_pull_request(cls, pr_id, comment=""):
         response = requests.get("https://api.github.com/repos/{}/pulls/{}".format(GlobalVars.bot_repo_slug, pr_id))
-        if not response:
+        files_list = requests.get("https://api.github.com/repos/{}/pulls/{}/files"
+                                  .format(GlobalVars.bot_repo_slug, pr_id))
+        if not response or not files_list:
             raise ConnectionError("Cannot connect to GitHub API")
         pr_info = response.json()
+        files_info = files_list.json()
         if pr_info["user"]["login"] != "SmokeDetector":
             raise ValueError("PR #{} is not created by me, so I can't approve it.".format(pr_id))
         if "<!-- METASMOKE-BLACKLIST" not in pr_info["body"]:
             raise ValueError("PR description is malformed. Blame a developer.")
         if pr_info["state"] != "open":
             raise ValueError("PR #{} is not currently open, so I won't merge it.".format(pr_id))
+        pr_string = pr_info["title"]
+        file_modified = files_info[0]["filename"]
         ref = pr_info['head']['ref']
+        requested_pr_string = regex.match(r".*?: \S+ (.*?)(?:\Z|\s*\(\?#)", pr_string).group(1)
```

Comment:
I'm not a huge fan of using a regex to parse the PR titles, because we might change the format, or we might have trouble with certain usernames or blacklist items. What about looking at the last entry of the changed blacklist file, or parsing the patch returned by the GitHub API?

Reply:
I think RegEx seems fine. Format changes -> the RegEx can change.